src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -414,13 +414,27 @@ extern "C" int32_t CryptoNative_GetSslConnectionInfo(SSL* ssl,
     return 0;
 }
 
+// Controls if ERR_clear_error is going to be called before SSL_write/SSL_read
+static bool g_forceOpenSslErrorQueueCleanupBeforeWriteRead = false;
+
+extern "C" void CryptoNative_ForceOpenSslErrorQueueCleanupBeforeWriteRead()
+{
+    g_forceOpenSslErrorQueueCleanupBeforeWriteRead = true;
+}
+
 extern "C" int32_t CryptoNative_SslWrite(SSL* ssl, const void* buf, int32_t num)
 {
+    if (g_forceOpenSslErrorQueueCleanupBeforeWriteRead)
```

Comment:
Does this check/branch have any impact on the case where it's not set? i.e. the default case where the user doesn't set the flag?  Presumably it's a drop in the bucket compared to the comparison, but it'd be good to know for sure.

Reply:
The performance with the config (default, optimization on) is indistinguishable compared to master - same test indicates that when the optimization is off there is a 5 to 15% extra time to complete the test (test based on your ConcurrentHttpsGet test, but I had to run it under Xunit since I couldn't get Benchmark.NET working against my local build).